src/main/java/com/elytradev/teckle/compat/ct/TeckleCTUtils.java

```diff
@@ -0,0 +1,160 @@
+package com.elytradev.teckle.compat.ct;
+
+import com.elytradev.concrete.reflect.accessor.Accessor;
+import com.elytradev.concrete.reflect.accessor.Accessors;
+import crafttweaker.CraftTweakerAPI;
+import crafttweaker.api.item.IIngredient;
+import crafttweaker.api.item.IItemStack;
+import crafttweaker.api.item.IngredientStack;
+import crafttweaker.api.minecraft.CraftTweakerMC;
+import crafttweaker.api.oredict.IOreDictEntry;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.oredict.OreDictionary;
+
+import java.util.List;
+
+/**
+ * Utilities and helper methods for CTAlloyFurnace.
+ * Used for comparing and converting between the various ways of representing ingredients.
+ */
+public class TeckleCTUtils {
+    private static Accessor<IIngredient> internalIngredient = Accessors.findField(IngredientStack.class, "ingredient");
```

Comment:
This shouldn't use reflection, make an issue on the CT tracker about IngredientStack#getInternal() returning null.

Reply:
I opened https://github.com/CraftTweaker/CraftTweaker/issues/631 since I agree that reflection is ugly, but it may be necessary if they don’t choose to change it.

Reply:
The crafttweaker dev requested the issue be opened, I doubt they'll ignore it.

Reply:
Ah, I didn’t realize that, I thought the null return was just deliberate encapsulation. In that case, I’ll change to use getInternal as soon as CT updates with the change.

Reply:
CraftTweaker just merged the PR fixing IngredientStack#getInternal, so I was able to remove the reflection like you asked.